compiler/rustc_metadata/src/creader.rs

```diff
@@ -414,10 +451,14 @@ impl<'a, 'tcx> CrateLoader<'a, 'tcx> {
 
         let extern_private = self.sess.opts.externs.get(name.as_str()).map(|e| e.is_private_dep);
 
+        if matches!(origin, CrateOrigin::Injected) {
+            return true;
+        }
```

Comment:
Maybe put this check above the `let extern_private = ...`?

Reply:
Done.